### DISCUSSION TITLE
Restriction of BT-109-Lot in case of duration > 4 years

### DISCUSSION THREAD

**Discussion by lesomue** [OP-TED/eForms-SDK#1068] [Q&A] [answered]
Hello, 

we have a question regarding the business rules which state that BT-109-Lot is forbidden in connection with a duration of 4 / 7 / 8 years. 

The fields.json states the following, which means that the execution period is only considered based on the specific duration in years, months etc. or as the difference between the start and end date. 

`"forbidden" : {
      "value" : false,
      "severity" : "ERROR",
      "constraints" : [ {
        "noticeTypes" : [ "1", "2", "3", "4", "5", "6", "14", "15", "19", "23", "24", "25", "26", "27", "28", "29", "30", "31", "32", "33", "34", "35", "36", "37", "38", "39", "40", "CEI", "T01", "T02", "X01", "X02" ],
        "value" : true,
        "severity" : "ERROR"
      }, {
        "noticeTypes" : [ "7", "10", "12", "16", "20" ],
        "condition" : "{ND-LotTenderingProcess} ${BT-765-Lot not in ('fa-mix','fa-w-rc','fa-wo-rc') or not(((BT-537-Lot - BT-536-Lot) > P4Y) or (BT-36-Lot > P4Y))}",
        "value" : true,
        "severity" : "ERROR"
      }, {
        "noticeTypes" : [ "8", "11", "13", "17", "21" ],
        "condition" : "{ND-LotTenderingProcess} ${BT-765-Lot not in ('fa-mix','fa-w-rc','fa-wo-rc') or not(((BT-537-Lot - BT-536-Lot) > P8Y) or (BT-36-Lot > P8Y))}",
        "value" : true,
        "severity" : "ERROR"
      }, {
        "noticeTypes" : [ "9", "18", "22" ],
        "condition" : "{ND-LotTenderingProcess} ${BT-765-Lot not in ('fa-mix','fa-w-rc','fa-wo-rc') or not(((BT-537-Lot - BT-536-Lot) > P7Y) or (BT-36-Lot > P7Y))}",
        "value" : true,
        "severity" : "ERROR"`

However, it is also possible to set the duration to `unlimited` via BT-538-Lot, which is logically speaking, longer than the specified 4 / 7 or 8 years. Nevertheless, the forbidden rule is applied when using `unlimited ` and the justification for the duration (BT-109-Lot) is forbidden. 

Is that the intenden validation behaviour? Actually, if the duration = unlimited, you would also have to specify a justification for the duration (BT-109), wouldn't you? 

Many thanks in advance and best regards
Lea Müller
cosinex GmbH

**Comment by YvesJo** [comment, marked as answer]
Hi,
I understand your concern and there is the need for some internal controls before I may answer positively to your question.
In the case this should be considered then we could expect a rule update with SDK 1.14. For previous versions it is still possible to specify a long duration (e.g. 99 years) instead or add the justification in a text field like Additional Information (BT-300).

**Comment by lesomue** [comment]
Okay, thank you for your quick reply!
